weblate/utils/requirements.py

```diff
@@ -169,10 +171,88 @@ def get_versions():
     return result
 
 
+def get_version_number_db(version):
+    """Returns version of database in format as (Major Version.Minor Version.Revision).
+
+    from an integer. To know more about database version,
+    refer <https://www.psycopg.org/docs/connection.html>.
+    """
+    major_version = version[0:2]
+    major_version = (
+        major_version.split(major_version[1])[0]
+        if major_version[1] == "0"
+        else major_version
+    )
+    minor_version = version[2:4] if version[2:4] != "00" else ""
+    if minor_version != "":
+        minor_version = (
+            minor_version.split(minor_version[1])[0]
+            if minor_version[1] == "0"
+            else minor_version
+        )
+    revision = version[4:]
+    return (
+        major_version + "." + minor_version + "." + revision
+        if minor_version != ""
+        else major_version + "." + revision
+    )
+
+
+def get_db_version():
+    if connection.vendor.upper() == "POSTGRESQL":
+        try:
+            result = (
+                connection.vendor,
+                "https://www.postgresql.org/",
+                get_version_number_db(
+                    str(connection.cursor().connection.server_version),
+                ),
+            )
+        except RuntimeError:
+            raise ImproperlyConfigured(
+                "Failed to get a database version. please install a database."
+            )
+    else:
+        try:
+            result = (
+                connection.vendor,
+                "https://www.mysql.com/",
+                connection.cursor().connection.get_server_info().split("-", 1)[0],
+            )
+        except RuntimeError:
+            raise ImproperlyConfigured(
+                "Failed to get a database version. please install a database."
+            )
+
+    return result
+
+
+def get_cache_version():
+    try:
+        if (
+            settings.CACHES.get("default").get("BACKEND")
```

Comment:
Chaining .get is probably not a good idea here - in case "default" is not present it will return None and next get will crash. So either rely on "default" being present or provide default value to get.

Reply:
Will fix this.

Reply:
`settings.CACHES.get("default", {}).get("BACKEND")` would work fine I guess.